src/devices/machine/dp83932c.cpp

```diff
@@ -572,3 +586,17 @@ void dp83932c_device::dump_bytes(u8 *buf, int length)
 				buf[i * 8 + 4], buf[i * 8 + 5], buf[i * 8 + 6], buf[i * 8 + 7]);
 	}
 }
+
+u16 dp83932c_device::read_bus_word(offs_t address)
+{
+	unsigned const width = (m_reg[DCR] & DCR_DW) ? 4 : 2;
+
+	return width == 2 ? m_bus->read_word(address) : m_bus->read_dword(address, 0xffff);
```

Comment:
I'll dig into this some more to see if I can find a better way to do this. While it has been a while since I wrote this part, from what I can remember, there is an endianness issue that is driving this. I tried switching it back and that caused MAME to segfault during NEWS-OS boot, so I'll need to trace through it again.

From what I remember, this was the diagram that led me to try this method out:
![image](https://user-images.githubusercontent.com/32155223/152631224-7ee4aa9d-98f1-41a4-8117-ba0af3f7d36f.png)

I'm probably missing something important here, so I'll shake off the rust and take another look at it :)

Reply:
Hmmm... maybe the problem is that I didn't convert it to use the full 32 bit data width. Changing it to this also works, and seems to align better with the above diagram. Thoughts?

```cpp
u32 dp83932c_device::read_bus_word(offs_t address)
{
	unsigned const width = (m_reg[DCR] & DCR_DW) ? 4 : 2;
	return width == 2 ? m_bus->read_word(address) : m_bus->read_dword(address);
}

void dp83932c_device::write_bus_word(offs_t address, u32 data)
{
	unsigned const width = (m_reg[DCR] & DCR_DW) ? 4 : 2;
	width == 2 ? m_bus->write_word(address, data) : m_bus->write_dword(address, data);
}
```

If I had to guess, the way I had it before just happened to work because it was only loading 16-bit values out of the 32-bit data path.

Reply:
Updated based on our offline conversation. Marking as resolved, but will split this change out into a separate PR.